### PR TITLE
Fix termination of processes in `FractalProcessPoolExecutor`

### LINKED ISSUE #1560: Review that threads/processes are closed properly by local-experimental runner
We should, at least once, verify that the new processpool-based backend closes all relevant threads and processes upon shutdown.
How to verify:
1. Start a fractal-server instance and add a single task with command pointing to a bash script with `sleep 10000` (or you can collect and use `fractal-tasks-mock`). See a script like tests/examples/ run_with_gunicorn.sh
2. Start a job which includes this task.
3. Check the active processes and threads (e.g. with `ps` - see also https://github.com/fractal-analytics-platform/fractal-containers/blob/main/slurm-stress/head/monitor_threads.sh - or maybe `top -H`).
4. Shutdown the job through the API and wait until it's marked as "failed"
5. Check again as in point 3.

---

Another interesting is to run N=100 very short jobs, and check (at the end) that the number of active threads/processes is under control.

### LINKED ISSUE #1568: Review compatibility of `local_experimental` backend runner with V1 jobs
1. The PR https://github.com/fractal-analytics-platform/fractal-server/pull/1544/files doesn't seem to include any V1-related code change or test.
2. Some test runs (https://github.com/fractal-analytics-platform/fractal-containers/actions/runs/9566318525/job/26371310670) seem to hang, when running V1 jobs.
3. Can we run a V1 job with the new backend runner?
    * If yes, let's add a test for it (or is there one already?)
    * If no, let's make the runner fail loudly.

### PR DESCRIPTION
**Pull request by ychiucco**
closes #1560 
closes #1568 

## Checklist before merging
- [x] I added an appropriate entry to `CHANGELOG.md`
- [ ] I merged `main` into the current branch.

### CODE REVIEW

File: fractal_server/app/runner/v2/_local_experimental/executor.py [hunk 1]
```diff
@@ -67,11 +66,15 @@ def _terminate_processes(self):
...
             for pid in self._processes.keys():
-                logger.debug(f"Sending SIGTERM to process {pid}")
-                os.kill(pid, signal.SIGTERM)
-                logger.debug(f"Process {pid} terminated.")
+                parent = psutil.Process(pid)
+                children = parent.children(recursive=True)
+                for child in children:
+                    child.kill()
+                parent.kill()
+                logger.debug(f"Process {pid} and its children terminated.")
```
Comment (tcompa) [inline fractal_server/app/runner/v2/_local_experimental/executor.py:77]: Do we see these logs anywhere?

File: CHANGELOG.md [hunk 2]
```diff
@@ -30,6 +30,8 @@
 * Dependencies
     * Add `fabric` to `dev` dependencies (\#1518).
     * Add new `postgres-psycopg-binary` extra (\#1562).
+* Runner
```
Comment (tcompa) [inline CHANGELOG.md:33]: move up

File: fractal_server/app/runner/v2/_local_experimental/executor.py (hunk 1 above)
Comment (ychiucco) [reply fractal_server/app/runner/v2/_local_experimental/executor.py]: We do, after 3b7c5d7c6c751c0b53860bc539174b2c959ade48

File: fractal_server/app/runner/v1/__init__.py [hunk 3]
```diff
@@ -91,22 +91,21 @@ async def submit_workflow(
...
-    else:
-        raise RuntimeError(f"Invalid runner backend {FRACTAL_RUNNER_BACKEND=}")
-
     with next(DB.get_sync_db()) as db_sync:
 
         job: ApplyWorkflow = db_sync.get(ApplyWorkflow, job_id)
         if not job:
             raise ValueError(f"Cannot fetch job {job_id} from database")
 
+        if FRACTAL_RUNNER_BACKEND in ["local", "local_experimental"]:
```
Comment (tcompa) [inline fractal_server/app/runner/v1/__init__.py:102]: This won't work.
If we map the new runner onto the old one, what would happen when the user tries to shut down a v1 job?

File: fractal_server/app/runner/v1/__init__.py [hunk 4]
```diff
@@ -53,6 +53,19 @@
 _backends["slurm"] = slurm_process_workflow
 
 
+def fail_job(*, db, job: ApplyWorkflow, log: str, logger_name: str) -> None:
+    logger = set_logger(logger_name=logger_name)
```
Comment (tcompa) [inline fractal_server/app/runner/v1/__init__.py:57]: this should be get_logger

File: fractal_server/app/runner/v2/__init__.py [hunk 5]
```diff
@@ -44,6 +45,19 @@
 _backends["slurm"] = slurm_process_workflow
 
 
+def fail_job(*, db, job: JobV2, log: str, logger_name: str) -> None:
+    logger = set_logger(logger_name=logger_name)
```
Comment (tcompa) [inline fractal_server/app/runner/v2/__init__.py:49]: this should be get_logger

File: fractal_server/app/runner/v2/__init__.py [hunk 6]
```diff
@@ -44,6 +45,19 @@
...
+    logger.error(log)
+    job.status = JobStatusTypeV2.FAILED
+    job.end_timestamp = get_timestamp()
+    job.log = log
+    db.merge(job)
+    db.commit()
+    db.close()
+    close_logger(logger)
```
Comment (tcompa) [inline fractal_server/app/runner/v2/__init__.py:57]: reset logger handlers, here

File: fractal_server/app/runner/v1/__init__.py [hunk 7]
```diff
@@ -53,6 +53,19 @@
...
+    logger.error(log)
+    job.status = JobStatusTypeV1.FAILED
+    job.end_timestamp = get_timestamp()
+    job.log = log
+    db.merge(job)
+    db.commit()
+    db.close()
+    close_job_logger(logger)
```
Comment (tcompa) [inline fractal_server/app/runner/v1/__init__.py:65]: reset logger handlers, here

File: fractal_server/app/runner/v1/__init__.py [hunk 8]
```diff
@@ -53,6 +53,19 @@
 _backends["slurm"] = slurm_process_workflow
 
 
+def fail_job(*, db, job: ApplyWorkflow, log: str, logger_name: str) -> None:
```
Comment (tcompa) [inline fractal_server/app/runner/v1/__init__.py:56]: add boolean argument `emit_log`

Comment (tcompa) [inline fractal_server/app/runner/v1/__init__.py:56]: add type hint for db

File: fractal_server/app/runner/v1/__init__.py [hunk 9]
```diff
@@ -147,9 +189,13 @@ async def submit_workflow(
         )
 
         if WORKFLOW_DIR_LOCAL.exists():
-            raise RuntimeError(
-                f"Workflow dir {WORKFLOW_DIR_LOCAL} already exists."
+            fail_job(
+                db=db_sync,
+                job=job,
+                log_msg=f"Workflow dir {WORKFLOW_DIR_LOCAL} already exists.",
+                logger_name=logger_name,
```
Comment (tcompa) [inline fractal_server/app/runner/v1/__init__.py:196]: i would also emit a log, here

File: fractal_server/app/runner/v2/__init__.py [hunk 10]
```diff
@@ -107,23 +140,18 @@ async def submit_workflow(
...
-            job.end_timestamp = get_timestamp()
-            job.log = f"Workflow dir {WORKFLOW_DIR_LOCAL} already exists."
-            db_sync.merge(job)
-            db_sync.commit()
-            db_sync.close()
+            fail_job(
+                db=db_sync,
+                job=job,
+                log_msg=f"Workflow dir {WORKFLOW_DIR_LOCAL} already exists.",
+                logger_name=logger_name,
```
Comment (tcompa) [inline fractal_server/app/runner/v2/__init__.py:153]: i would also emit a log, here

File: fractal_server/app/runner/v2/__init__.py [hunk 11]
```diff
@@ -167,15 +195,15 @@ async def submit_workflow(
...
-                "An error occurred while creating job folder and subfolders.\n"
-                f"Original error: {str(e)}"
+            fail_job(
+                db=db_sync,
+                job=job,
+                log_msg=(
+                    "An error occurred while creating job folder and "
+                    f"subfolders.\nOriginal error: {str(e)}"
+                ),
+                logger_name=logger_name,
```
Comment (tcompa) [inline fractal_server/app/runner/v2/__init__.py:205]: i would also emit a log, here

**Pull request merged**